seacrowd/sea_datasets/mswc/mswc.py

```diff
@@ -0,0 +1,219 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+import pandas as pd
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import TASK_TO_SCHEMA, Licenses, Tasks
+
+_CITATION = """\
+@inproceedings{NEURIPS DATASETS AND BENCHMARKS2021_fe131d7f,
+    author = {Mazumder, Mark and Chitlangia, Sharad and Banbury, Colby and Kang, Yiping and Ciro, Juan and Achorn, Keith and Galvez,
+    Daniel and Sabini, Mark and Mattson, Peter and Kanter, David and Diamos, Greg and Warden, Pete and Meyer, Josh and Janapa Reddi,
+    Vijay},
+    booktitle = {Proceedings of the Neural Information Processing Systems Track on Datasets and Benchmarks},
+    editor = {J. Vanschoren and S. Yeung},
+    pages = {},
+    publisher = {Curran},
+    title = {Multilingual Spoken Words Corpus},
+    url = {https://datasets-benchmarks-proceedings.neurips.cc/paper_files/paper/2021/file/fe131d7f5a6b38b23cc967316c13dae2-Paper-round2.pdf},
+    volume = {1},
+    year = {2021}
+}
+"""
+
+_DATASETNAME = "mswc"
+
+_DESCRIPTION = """\
+Multilingual Spoken Words Corpus is a large and growing audio dataset of spoken words in 50 languages collectively spoken by over 5 billion people, for academic research and commercial applications in keyword spotting and spoken term search.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/MLCommons/ml_spoken_words"
+
+_LANGUAGES = ["cnh", "ind", "vie"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+_LANGUAGE_NAME_MAP = {
+    "cnh": "cnh",
+    "ind": "id",
+    "vie": "vi",
+}
+
+_FORMATS = ["wav", "opus"]
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_LOCAL = False
+
+_URLS = "https://huggingface.co/datasets/MLCommons/ml_spoken_words/resolve/refs%2Fconvert%2Fparquet/{lang}_{format}/{split}/0000.parquet?download=true"
+
+_SUPPORTED_TASKS = [Tasks.SPEECH_RECOGNITION]
+_SUPPORTED_SCHEMA_STRINGS = [f"seacrowd_{str(TASK_TO_SCHEMA[task]).lower()}" for task in _SUPPORTED_TASKS]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class MSWC(datasets.GeneratorBasedBuilder):
+    """
+    Multilingual Spoken Words Corpus is a large and growing audio dataset of spoken words in 50 languages collectively spoken by over 5 billion people, for academic research and commercial applications in keyword spotting and spoken term search.
+    """
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = []
+
+    for language in _LANGUAGES:
+        for format in _FORMATS:
+            subset_id = f"{language}_{format}"
+            BUILDER_CONFIGS.append(
+                SEACrowdConfig(name=f"{subset_id}_source", version=SOURCE_VERSION, description=f"{_DATASETNAME} source schema", schema="source", subset_id=subset_id),
```

Comment:
ugh missed this in the earlier review. ideally the config name should be prefixed with `_DATASETNAME`, both applied to `source` and `seacrowd`

and dont' forget to adjust your code accordingly when extracting the lang and format

```suggestion
                SEACrowdConfig(name=f"{_DATASETNAME}_{subset_id}_source", version=SOURCE_VERSION, description=f"{_DATASETNAME} source schema", schema="source", subset_id=subset_id),
```